js/id/id.js

```diff
@@ -350,11 +372,22 @@ iD.version = '1.7.0';
         return navigator.userAgent.indexOf(x) !== -1;
     }
 
-    if (nav('Win')) detected.os = 'win';
-    else if (nav('Mac')) detected.os = 'mac';
-    else if (nav('X11')) detected.os = 'linux';
-    else if (nav('Linux')) detected.os = 'linux';
-    else detected.os = 'win';
+    if (nav('Win')) {
+        detected.os = 'win';
+        detected.platform = 'Windows';
+    }
+    else if (nav('Mac')) {
+        detected.os = 'mac';
+        detected.platform = 'Macintosh';
+    }
+    else if (nav('X11') || nav('Linux')) {
+        detected.os = 'linux';
+        detected.platform = 'Linux';
+    }
+    else {
+        detected.os = 'win';
```

Comment:
`detected.os` is used for only one thing in iD, and that is to setup the keyboard shortcuts in [cmd.js](https://github.com/openstreetmap/iD/blob/master/js/id/ui/cmd.js).  

I suppose `win` was just picked as a conservative default for the keyboard shortcut setup.  In reality, if the code runs here, the user is probably on a device without a keyboard - like a game console or an Android phone.